File: app/src/main/java/apps/sai/com/imageresizer/myimages/MyImagesPresenter.java
package apps.sai.com.imageresizer.myimages;

import android.content.Context;
import android.content.Intent;

import java.util.List;

import apps.sai.com.imageresizer.data.DataApi;
import apps.sai.com.imageresizer.data.FileApi;
import apps.sai.com.imageresizer.data.ImageInfo;

/**
 * Created by sailesh on 20/01/18.
 */

public class MyImagesPresenter implements MyImagesContract.Presenter {

    MyImagesContract.View mView;
    @Override
    public void takeView(MyImagesContract.View view) {
        mView =view;
    }

    @Override
    public void launchgalleryExternalApp(boolean singlePhoto) {

    }

    @Override
    public void onGalleryImageSelected(Intent data) {

    }

    @Override
    public void dropView() {
    mView =null;
    }

    @Override
    public void showSelectedImage(ImageInfo imageInfo) {

    }

    @Override
    public void deleteImage(ImageInfo imageInfo) {

    }

    @Override
    public void shareImages(Context context, List<ImageInfo> imageInfoList) {

    }

    @Override
    public List<ImageInfo> getImages(Context context) {

        DataApi dataApi = new FileApi(context);
        List<ImageInfo> imageInfoList = dataApi.getMyImages();

//        for (int i = 0; i <imageInfoList.size() ; i++) {
//            ImageInfo imageInfo = imageInfoList.get(i);
//
//        }

        return imageInfoList;


    }
}

File: app/src/main/java/apps/sai/com/imageresizer/myimages/MyImagesPresenter.kt
package apps.sai.com.imageresizer.myimages

import android.content.Context
import android.content.Intent
import apps.sai.com.imageresizer.data.DataApi
import apps.sai.com.imageresizer.data.FileApi
import apps.sai.com.imageresizer.data.ImageInfo
import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.flow.Flow
import kotlinx.coroutines.flow.flow
import kotlinx.coroutines.flow.flowOn

/**
 * Created by sailesh on 20/01/18.
 */
class MyImagesPresenter : MyImagesContract.Presenter {
    private var mView: MyImagesContract.View? = null
    override fun takeView(view: MyImagesContract.View) {
        mView = view
    }

    override fun launchgalleryExternalApp(singlePhoto: Boolean) {
    }

    override fun onGalleryImageSelected(data: Intent) {
    }

    override fun dropView() {
        mView = null
    }

    override fun showSelectedImage(imageInfo: ImageInfo) {
    }

    override fun deleteImage(imageInfo: ImageInfo) {
    }

    override fun shareImages(context: Context, imageInfoList: List<ImageInfo>) {
    }

    override fun getImages(context: Context): Flow<List<ImageInfo>> = flow {
        val dataApi: DataApi = FileApi(context)
        val imageInfoList = dataApi.myImages
        emit(imageInfoList)
    }.flowOn(Dispatchers.IO)

}
